sor/src/main/java/com/bazaarvoice/emodb/sor/db/DeltaIterator.java

```diff
@@ -0,0 +1,201 @@
+        UUID correctChangeId = getChangeId(_next);
+
+        if (_list == null) {
+            _list = Lists.newArrayListWithCapacity(3);
+        }
+
+        _list.add(_next);
+
+
+        _oldDelta = _next;
+
+        while (_iterator.hasNext() && getChangeId(_next = _iterator.next()).equals(correctChangeId)) {
+            _list.add(_next);
+            contentSize += getValue(_next).remaining();
+            _next = null;
+        }
+
+        Collections.reverse(_list);
+
+        // delta is fragmented if the first block is not zero. In this case, we skip it.
+        if (getBlock(_list.get(0)) != 0) {
+            return null;
+        }
+
+        int numBlocks = getNumBlocks(_list.get(0));
+
+        while (numBlocks != _list.size()) {
+            contentSize -= getValue(_list.remove(_list.size() - 1)).remaining();
+        }
+
+        return stitchContent(contentSize);
+
+    }
+
+    // stitch delta together and return it as one bytebuffer
+    private ByteBuffer compute(int numBlocks) {
+
+        int contentSize = getValue(_next).remaining();
+        _oldDelta = _next;
+        UUID changeId = getChangeId(_next);
+
+        if (_list == null) {
+            _list = Lists.newArrayListWithCapacity(numBlocks);
+        }
+
+        _list.add(_next);
+
+        for (int i = 1; i < numBlocks; i++) {
+            if (_iterator.hasNext()) {
+                _next = _iterator.next();
+                if (getChangeId(_next).equals(changeId)) {
+                    _list.add(_next);
+                    contentSize += getValue(_next).remaining();
+                } else {
+                    // fragmented delta encountered, we must skip over it
+                    return null;
+                }
+            } else {
+                // fragmented delta and no other deltas to skip to
+                throw new DeltaStitchingException();
+            }
+        }
+
+        skipForward();
+
+        return stitchContent(contentSize);
+    }
+
+    @Override
+    protected T computeNext() {
+
+        while(true) {
+
+            if (_next == null) {
+                return endOfData();
+            }
+
+            ByteBuffer content;
+
+            if (!_reverse) {
+
+                int numBlocks = getNumBlocks(_next);
+
+                if (numBlocks == 1) {
+                    T ret = convertDelta(_next);
+                    skipForward();
+                    return ret;
+                }
+
+                content = compute(numBlocks);
+
+            } else {
+                if (getBlock(_next) == 0) {
+                    if (getNumBlocks(_next) != 1) {
+                        continue;
+                    }
+                    T ret = convertDelta(_next);
+                    _next = _iterator.hasNext() ? _iterator.next() : null;
+                    return ret;
+                }
+
+                content = reverseCompute();
+            }
+
+            _list.clear();
+
+            if (content == null) {
```

Comment:
Redundant `continue` since there is only a single line after this `if` statement.  Cleaner to invert, `if (content != null) { return convertDelta(_oldDelta, content); }`

Reply:
done in migration

Reply:
done